src/main/java/main/UserInput.java

```diff
@@ -8,170 +8,207 @@
+     */
+    public UserInput(String[] args) {
+        this.args = args;
+    }
+
+    /**
+     * Parse the command line arguments
+     * @return The user request read from the command line arguments
+     * @throws ParseException Thrown in caes the command line arguments are invalid
+     */
+    public Request parse() throws ParseException {
+        CommandLineParser dp = new DefaultParser();
+        commandLine = dp.parse(UserInputOption.getOptions(), args, false);
+
+        if (commandLine.hasOption(UserInputOption.HELP.getOption().getOpt())) {
+            return Request.HELP;
+        }
+        if (commandLine.hasOption(UserInputOption.VERSION.getOption().getOpt())) {
+            return Request.VERSION;
+        }
+
+        // "gui" and "file" options are mutually exclusive
+        if (commandLine.hasOption(UserInputOption.GUI.getOption().getOpt())
+            && commandLine.hasOption(UserInputOption.FILE.getOption().getOpt())) {
+            throw new ParseException("Gui and file option cannot be used at the same time.");
+        } else {
+            return Request.GENERATE;
+        }
+    }
+
+    /**
+     * Print the command line help
+     */
+    public void printHelp() {
+        HelpFormatter formatter = new HelpFormatter();
+        formatter.printHelp("TimeSheetGenerator", UserInputOption.getOptions());
+    }
+
+    /**
+     * Print the version of the application.
+     * If possible additional build information is printed as well.
+     */
+    public void printVersion() {
+        String version = null;
+        String buildTime = null;
+        String branch = null;
+        String commit = null;
+
+        // read the project properties from the resources
+        try {
+            InputStream inputStream = Main.class.getClassLoader().getResourceAsStream("project.properties");
+
+            if (inputStream != null) {
+                // parse the properties
+                Properties properties = new Properties();
+                properties.load(inputStream);
+
+                // the build properties are encoded in the buildInfo property
+                String[] buildProperties = properties.getProperty("buildInfo").split(";");
+                if (buildProperties.length >= 4) {
+                    version = buildProperties[0];
+                    // if a string starts with a "$" the placeholder has not been replaced by the actual value
+                    buildTime = buildProperties[1];
+                    if (buildTime.startsWith("$")) {
+                        buildTime = null;
+                    }
+                    branch = buildProperties[2];
+                    if (branch.startsWith("$")) {
+                        branch = null;
+                    }
+                    commit = buildProperties[3];
+                    if (commit.startsWith("$")) {
+                        commit = null;
+                    }
+                }
+            }
+        } catch (IOException e) {}
```

Comment:
Oh, I think we forgot to fill this catch block here? Not part of this PR, nevertheless, fixing that would be great!

Reply:
This is left out on purpose. The relevant variables are set to null before the try-catch block, so if none of them is set, the error will be handled in the if-clause below.